packages/ra-core/src/routing/useRedirect.ts

```diff
@@ -1,5 +1,6 @@
 import { useCallback } from 'react';
 import { useNavigate, To } from 'react-router-dom';
+import { Location } from 'history';
```

Comment:
Why don't you import the `To` type imported from 'react-router-dom' above? It would avoid the added dependency